kano/network.py

```diff
@@ -535,6 +546,75 @@ def reload_kernel_module(device_vendor='148f', device_product='5370', module='rt
     return reloaded
 
 
+def connection_result(iface, wpa_file, connection_timeout, verbose=False):
+    '''
+    This function connects to the WPA supplicant event flow to detect the result of a connection attempt.
+    It returns a return code with an indication of the connection result. See the RC_* constants.
+
+    This code uses heuristics based on the supplicant source code: https://w1.fi/wpa_supplicant/
+    '''
+    rc=None
+    connected=False
+    scans = 0
+
+    def is_internet_up(monitor_file=INTERNET_UP_FILE):
+        return os.path.isfile(monitor_file)
+
+    run_cmd("wpa_supplicant -D nl80211,wext -t -d -c%s -i%s -f /var/log/kano_wpa.log -B" % (wpa_file, iface))
```

Comment:
Don't you want to use your `SUPPLICANT_CMD` template above? And also use `SUPPLICANT_LOGFILE`?

Reply:
Ups, missed it. Fixed